src/test/java/bio/terra/integration/FileTest.java

```diff
@@ -0,0 +1,147 @@
+
+    @Autowired
+    private Storage storage;
+
+    private StudySummaryModel studySummaryModel;
+    private String studyId;
+
+    @Before
+    public void setup() throws Exception {
+        super.setup();
+        studySummaryModel = dataRepoFixtures.createStudy(steward(), "file-acl-test-study.json");
+        studyId = studySummaryModel.getId();
+        logger.info("created study " + studyId);
+        dataRepoFixtures.addStudyPolicyMember(
+            steward(), studySummaryModel.getId(), SamClientService.DataRepoRole.CUSTODIAN, custodian().getEmail());
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (studyId != null) {
+            dataRepoFixtures.deleteStudy(steward(), studyId);
+        }
+    }
+
+    @Test
+    public void fileUnauthorizedPermissionsTest() throws Exception {
+
+        String gsPath = "gs://" + testConfiguration.getIngestbucket();
+        String filePath = "/foo/bar";
+
+        DataRepoResponse<JobModel> launchResp = dataRepoFixtures.ingestFileLaunch(
+            custodian(), studyId, gsPath + "/files/File%20Design%20Notes.pdf", filePath);
+        assertThat("Custodian is not authorized to ingest a file",
+            launchResp.getStatusCode(),
+            equalTo(HttpStatus.UNAUTHORIZED));
+
+        FSObjectModel fsObjectModel = dataRepoFixtures.ingestFile(
+            steward(), studyId, gsPath + "/files/File%20Design%20Notes.pdf", filePath);
+        String fileId = fsObjectModel.getObjectId();
+
+        String json = String.format("{\"file_id\":\"foo\",\"file_ref\":\"%s\"}", fileId);
+
+        String targetPath = "scratch/file" + UUID.randomUUID().toString() + ".json";
+        BlobInfo targetBlobInfo = BlobInfo
+            .newBuilder(BlobId.of(testConfiguration.getIngestbucket(), targetPath))
+            .build();
+
+        try (WriteChannel writer = storage.writer(targetBlobInfo)) {
+            writer.write(ByteBuffer.wrap(json.getBytes(StandardCharsets.UTF_8)));
+        }
+
+        IngestResponseModel ingestResponseModel = dataRepoFixtures.ingestJsonData(
+            steward(), studyId, "file", targetPath);
+
+        assertThat("1 Row was ingested", ingestResponseModel.getRowCount(), equalTo(1L));
+
+        dataRepoFixtures.getFileById(steward(), studyId, fileId);
+        dataRepoFixtures.getFileById(custodian(), studyId, fileId);
+
+        DataRepoResponse<FSObjectModel> getResp = dataRepoFixtures.getFileByIdRaw(reader(), studyId, fileId);
+        assertThat("Reader is not authorized to get a file from a study",
+            getResp.getStatusCode(),
+            equalTo(HttpStatus.UNAUTHORIZED));
+
+        // get file by id
+        DataRepoResponse<FSObjectModel> fileResp = dataRepoFixtures.getFileByIdRaw(discoverer(), studyId, fileId);
+        assertThat("Discoverer is not authorized to get a file from a study",
+            fileResp.getStatusCode(),
+            equalTo(HttpStatus.UNAUTHORIZED));
+
+        // get file by name
```

Comment:
maybe update this comment so that it captures the fact that getFileByName is asserting that these roles can successfully get the files?